### PR TITLE
Duplicateof view

### PR DESCRIPTION
**Pull request by ZiyunShang**
### Purpose

This PR add 2 nodes for Duplicate:
DuplicateView SheetNodes

#### UnitTests
DuplicateView SheetNodesTests

#### SystemTests
DuplicateView SheetSystemTests

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] Snapshot of UI changes, if any.

### Reviewers

@AndyDu1985 @wangyangshi @mjkkirschner @QilongTang 

### FYIs

Screenshots of the dynamo graphs for the system test
#### DuplicateView
DuplicateView_2020-07-08_02-34-27

#### DuplicateSheet
DuplicateSheet_2020-07-08_02-34-54

### CODE REVIEW

File: src/Libraries/RevitNodes/Elements/Views/Sheet.cs [hunk 1]
```diff
@@ -506,6 +521,110 @@ public static Sheet ByNameNumberTitleBlock(string sheetName, string sheetNumber,
...
+        /// <param name="prefix"></param>
+        /// <param name="suffix"></param>
+        /// <returns></returns>
+        public static Sheet DuplicateSheet(Sheet sheet, int viewDuplicateOption = 0, string prefix = "", string suffix = "")
+        {            
+            if (sheet == null)
+                throw new ArgumentNullException(nameof(sheet));
+
+            ViewDuplicateOption Option = 0;
+            switch (viewDuplicateOption)
```
Comment (AndyDu1985) [inline src/Libraries/RevitNodes/Elements/Views/Sheet.cs:538]: Need to check with QA if we need the option for Sheet, or using `ViewDuplicateOption.WithDetailing` by default.

File: src/Libraries/RevitNodes/Elements/Views/Sheet.cs [hunk 2]
```diff
@@ -506,6 +521,110 @@ public static Sheet ByNameNumberTitleBlock(string sheetName, string sheetNumber,
...
+        /// Duplicates A Sheet. 
+        /// </summary>
+        /// <param name="sheet">The Sheet to be Duplicated</param>
+        /// <param name="viewDuplicateOption">Enter View Duplicate Option: 0 = Duplicate. 1 = AsDependent. 2 = WithDetailing.</param>
+        /// <param name="prefix"></param>
+        /// <param name="suffix"></param>
+        /// <returns></returns>
+        public static Sheet DuplicateSheet(Sheet sheet, int viewDuplicateOption = 0, string prefix = "", string suffix = "")
+        {            
+            if (sheet == null)
```
Comment (AndyDu1985) [inline src/Libraries/RevitNodes/Elements/Views/Sheet.cs:534]: The `prefix `and `suffix` is both for view and sheet? I think we should have them seperate for view and sheet.

Comment (ZiyunShang) [reply src/Libraries/RevitNodes/Elements/Views/Sheet.cs:534]: Do you mean that this node should have 2 prefix and suffix?

File: src/Libraries/RevitNodes/Elements/Views/Sheet.cs [hunk 3]
```diff
@@ -506,6 +521,110 @@ public static Sheet ByNameNumberTitleBlock(string sheetName, string sheetNumber,
...
+                    {
+                        elementIds.Add(element.Id);
+                    }
+                    Autodesk.Revit.UI.UIDocument uIDocument = new Autodesk.Revit.UI.UIDocument(Document);
+                    var openedViews = uIDocument.GetOpenUIViews().ToList();
+                    var shouldClosedViews = openedViews.FindAll(x => elementIds.Contains(x.ViewId));
+                    if (shouldClosedViews.Count > 0)
+                    {
+                        foreach (var v in shouldClosedViews)
+                            v.Close();
```
Comment (AndyDu1985) [inline src/Libraries/RevitNodes/Elements/Views/Sheet.cs:579]: To check if the view is the last view, if yes, delete those views that all ready closed.

Comment (ZiyunShang) [reply src/Libraries/RevitNodes/Elements/Views/Sheet.cs:579]: So delete all view but last one, do we need to throw exception if there is a last view?

**Pull request merged**